addons/web/static/src/js/core/popover.js

```diff
@@ -22,7 +22,7 @@ odoo.define('web.Popover', function () {
         constructor() {
             super(...arguments);
             this.popoverRef = useRef('popover');
-            this.orderedPositions = ['top', 'right', 'bottom', 'left'];
+            this.orderedPositions = ['top', 'top-end', 'top-start', 'end', 'bottom', 'bottom-right', 'bottom-start', 'left'];
```

Comment:
Still need to clean this up and review PR and what we want as a whole, but quick comment just to note `end` is supposed to be `bottom-end` here.